### ISSUE TITLE
Support partition_on on timestamp columns

### ISSUE THREAD

**Issue by wangxiaoying** [sfu-db/connector-x#313]
#### Describe your feature request

Support partition_on on timestamp columns https://github.com/sfu-db/connector-x/issues/305

**Comment by mvanaltvorst** [comment]
Hi! Is anyone working on this?

**Comment by wangxiaoying** [comment]
Hi @mvanaltvorst , thanks for the reply. Currently the logic of partitioning the numeric columns is [here](https://github.com/sfu-db/connector-x/blob/main/connectorx/src/partition.rs). Basically there are three steps:
1. get min and max value for partition column [get_col_range](https://github.com/sfu-db/connector-x/blob/main/connectorx/src/partition.rs#L89)
2. compute the intervals for query, this part of the logic can be found [here](https://github.com/sfu-db/connector-x/blob/main/connectorx/src/partition.rs#L64)
3. generate partitioned queries [get_part_query](https://github.com/sfu-db/connector-x/blob/main/connectorx/src/partition.rs#L108)

Basically we need to be able to support the three steps for timestamp columns too. Please feel free to open a PR for this if you are interested!

**Comment by surister** [comment]
For this issue to be considered accepted, should this be implemented in all databases?

**Comment by wangxiaoying** [comment]
Hi @surister , yes I think that would be the ideal case.
